app/house.py

```diff
@@ -0,0 +1,167 @@
+                    }
+                )
+        elif decoded_token["token_type"] == "generic_join_house":
+            db_user.houses.append(House.query.get(decoded_token["house_id"]))
+            db.session.commit()
+            return jsonify(
+                {
+                    "msg": "You are now in this house.",
+                    "data": decoded_token["house_id"],
+                    "status": "success",
+                }
+            )
+    except jwt.DecodeError:
+        return (
+            jsonify(msg="The supplied token was invalid.", status="error", data=""),
+            400,
+        )
+
+
+@house_blueprint.route("/<int:house_id>/user/invite/<string:identifier>")
+def specific_email(house_id, identifier):
+    house = House.query.get(house_id)
+    if not house.users.filter(
+        (User.username == identifier) | (User.email == identifier)
+    ).first():
+        return jsonify(
+            {
+                "status": "error",
+                "msg": "You don't have permission to do this",
+                "data": "",
+            }
+        )
+    user = User.filter(
+        (User.username == identifier) | (User.email == identifier)
+    ).first()
+    if not user:
+        return (
+            jsonify(
+                {"msg": "That user could not be found.", "status": "error", "data": ""}
+            ),
+            404,
+        )
+    if house:
+        token = jwt.encode(
+            {
+                "token_type": "specific_join_house",
+                "house_id": house.id,
+                "user_id": user.id,
+            },
+            current_app.config["SECRET_KEY"],
+        )
+        return jsonify(
+            {
+                "msg": "Successfully created an invite link",
+                "status": "success",
+                "data": token,
+            }
+        )
+    else:
+        return (
+            jsonify(
+                status="error", msg="A house with that ID cannot be found", data=""
+            ),
+            404,
+        )
+
+
+@house_blueprint.route("/user")
+@jwt_required
+def all_user_houses():
+    user = get_jwt_identity()
+    db_user: User = User.query.get(user)
+    return_data = jsonify(
+        {
+            "data": list(map(lambda x: x.id, db_user.houses)),
+            "msg": "",
+            "status": "success",
+        }
+    )
+    return return_data
+
+
+@house_blueprint.route("/<int:house_id>/user/invite")
+def generic_invite(house_id):
+    house = House.query.get(house_id)
+    if house:
+        token = url_for(
+            "house.join_house",
+            token=jwt.encode(
+                {"token_type": "generic_join_house", "house_id": house.id},
+                current_app.config["SECRET_KEY"],
+            ),
+        )
+        return jsonify(data=token, msg="", status="success")
+    else:
+        return (
+            jsonify(
+                status="error", msg="A house with that ID cannot be found.", data=""
```

Comment:
Data: `house_id`

Reply:
This is an error message –  so no data is returned.